### PR TITLE
Ee 3842/self assessment one year

### PR DESCRIPTION
**Pull request by CameronDowner**

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/proving/income/api/SelfAssessmentThresholdCalculator.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        return ONE_DEPENDANT_THRESHOLD.add(thresholdIncrement);
+    }
+
+    private void validateDependants(Integer dependants) {
+        if (dependants < 0) {
+            throw new IllegalArgumentException(String.format("Number of dependants cannot be less than zero, %d given", dependants));
+        }
+    }
+}
```
Comment (mikehalmaho) [inline src/main/java/uk/gov/digital/ho/proving/income/api/SelfAssessmentThresholdCalculator.java]: I'm not sure why we need a new threshold calculator - the old one does the same job with the yearly method?

**Review by DrewCrook — APPROVED** [review]
All fine. Andy Crook.

File: src/test/java/uk/gov/digital/ho/proving/income/validator/CatBNonSalariedIncomeValidatorTest.java [hunk 2]
```diff
@@ -10,25 +14,56 @@
...
-    private CatBNonSalariedIncomeValidator validator = new CatBNonSalariedIncomeValidator();
+    @Mock
+    private EmploymentCheckIncomeValidator employmentCheckIncomeValidator;
+
+    private CatBNonSalariedIncomeValidator validator;
+
+    @Before
+    public void setUp() {
+        validator = new CatBNonSalariedIncomeValidator(employmentCheckIncomeValidator);
+    }
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/proving/income/validator/CatBNonSalariedIncomeValidatorTest.java]: Couldn't you use @InjectMocks to do this for you?

File: src/test/java/uk/gov/digital/ho/proving/income/validator/CatBNonSalariedIncomeValidatorTest.java [hunk 3]
```diff
@@ -10,25 +14,56 @@
...
+        employmentCheckPasses();
+
         LocalDate raisedDate = getDate(2018, Month.SEPTEMBER, 23);
         List<ApplicantIncome> incomes = singleMonthlyPaymentAboveNoDependantsThreshold(raisedDate);
 
         IncomeValidationRequest request = new IncomeValidationRequest(incomes, raisedDate, 0);
         IncomeValidationResult result = validator.validate(request);
 
-        assertThat(result.calculationType()).isEqualTo(CatBNonSalariedIncomeValidator.CALCULATION_TYPE)
+        assertThat(result.calculationType()).isEqualTo("Category B non salaried")
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/proving/income/validator/CatBNonSalariedIncomeValidatorTest.java:62]: Can we discuss this?

File: src/test/java/uk/gov/digital/ho/proving/income/validator/IncomeValidatorTestIT.java [hunk 4]
```diff
@@ -22,13 +24,7 @@
...
 @RunWith(SpringRunner.class)
 @SpringBootTest(classes = {
-    IncomeValidationService.class,
-    CatASalariedIncomeValidator.class,
-    CatASalariedMonthlyIncomeValidator.class,
-    CatASalariedWeeklyIncomeValidator.class,
-    CatAUnsupportedIncomeValidator.class,
-    CatBNonSalariedIncomeValidator.class,
-    EmploymentCheckIncomeValidator.class
+    ServiceRunner.class
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/proving/income/validator/IncomeValidatorTestIT.java]: If you want to bring up the entire application I believe you can omit classes entirely... I'm unclear of the pros for doing so, can we discuss?

File: src/test/java/uk/gov/digital/ho/proving/income/validator/domain/TaxYearTest.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
+package uk.gov.digital.ho.proving.income.validator.domain;
+
+import org.junit.Test;
+
+import java.time.*;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
```
Comment (mikehalmaho) [inline src/test/java/uk/gov/digital/ho/proving/income/validator/domain/TaxYearTest.java]: Any reason this uses the Java 6 assertions?

**Review by mikehalmaho — CHANGES_REQUESTED** [review]

**Review by mikehalmaho — APPROVED** [review]

**Pull request merged**